cave/plot/configurator_footprint.py

```diff
@@ -355,131 +371,174 @@ def get_conf_matrix(self):
         return np.array(conf_matrix), conf_list, runs_runs_conf
 
     def _get_size(self, r_p_c):
-        return 10 + ((r_p_c - self.min_runs_per_conf) / (self.max_runs_per_conf - self.min_runs_per_conf)) * 40
-
-    def plot(self, X, conf_list: list, runs_runs_conf, inc_list: list,
-             runs_labels=None, contour_data=None):
-        '''
-            plots sampled configuration in 2d-space;
-            saves results in self.output, if set
+        sizes = 10 + ((r_p_c - self.min_runs_per_conf) / (self.max_runs_per_conf - self.min_runs_per_conf)) * 40
+        sizes *= np.array([0 if r == 0 else 1 for r in r_p_c])  # 0 size if 0 runs
+        return sizes
 
-            Parameters
-            ----------
-            X: np.array
-                np.array with 2-d coordinates for each configuration
-            conf_list: list
-                list of configurations in the same order as X
-            runs_runs_conf: list[np.array]
-                list of runs to be analyzed, each a np.array with runs per
-                config. if 2 runs are analyzed with 3 configs, this might look
-                like: [np.array([0,2,1]), np.array([2,7,1])]
-            inc_list: list
-                list of incumbents (Configuration)
-            contour_data: list
-                contour data (xx,yy,Z)
+    def _get_zorder(self, cds):
```

Comment:
I'm not convinced this is the right way to handle the z-order.
This way, a random configuration that was evaluated often might be (completely) overlapped by some other random configurations that were not evaluated as often. And we want to highlight the configurations that were evaluated most often.
If we set the z-order according to size, the configurations that were evaluated the most will be on top and show that this area was evaluated many times.

Reply:
The problem here is that bokeh supports zorder only the order of the drawing, so we need to iterate over the zorder-levels and scatter them after each other. Not really a problem for a finite number, I will think of a more elegant way.

Reply:
I introduced a new system, maybe you can have a look? The idea is to use bins for the sizes, so smaller dots will generally be under bigger dots and how fine that control system is can be determined by the number of bins used.